android/app/src/main/java/io/mosip/registration_client/api_services/AuthenticationApi.java

```diff
@@ -0,0 +1,194 @@
+package io.mosip.registration_client.api_services;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.mosip.registration.clientmanager.dto.http.ResponseWrapper;
+import io.mosip.registration.clientmanager.dto.http.ServiceError;
+import io.mosip.registration.clientmanager.exception.InvalidMachineSpecIDException;
+import io.mosip.registration.clientmanager.service.LoginService;
+import io.mosip.registration.clientmanager.spi.AuditManagerService;
+import io.mosip.registration.clientmanager.spi.SyncRestService;
+import io.mosip.registration.clientmanager.util.SyncRestUtil;
+import io.mosip.registration_client.R;
+import io.mosip.registration_client.model.AuthResponsePigeon;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+@Singleton
+public class AuthenticationApi implements AuthResponsePigeon.AuthResponseApi {
+    private Context context;
+    AuthResponsePigeon.AuthResponse authResponse;
```

Comment:
why are we creating a class level variable to hold the response, it should be method scoped

Reply:
Changed the variable scope to method level